### ISSUE TITLE
[JENKINS-20620] Memory Leak on Jenkins LTS 1.509.4/1.532.1

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#7176]
<p>After Jenkins upgraded to LTS 1.509.4 from LTS 1.509.3 I notice that over time (24 hours) Jenkins becomes very slowly. <br/>
It turns out that Jenkins (under Jetty service) slowly "eats" the server memory. It's takes about 24 hours to take all the memory allocated to Jenkins (42GB). See the snapshots with examples...</p>

<p>TEST #1 on LTS 1.509.4:<br/>
1. Machine with Jetty up after restart<br/>
2. Jenkins Used - After one Hour: 22GB<br/>
3. Jenkins Used - After 12 Hours: 27GB<br/>
4. Jenkins Used - After 20 Hours: 35GB -> Memory leaks between 10:00-10:20 as you can see after GC it's still think Java in-use and fail to cleanup the all memory as it should be.<br/>
5. Jenkins Used - After 23 Hours: 39GB -> Very slow response and Heap is almost 100%</p>

<p>TEST #2 on LTS 1.509.4:<br/>
I tried to do manual GC, Doesn't help!<br/>
(see attach file: "Monitor_Memory_Over_Time_Manual_GC")</p>

<p>TEST #3 on LTS 1.509.3:<br/>
Unfortunately I downgrade to LTS 1.509.3 because of the memory leak, for me it's a blocker issue! </p>

<p>Please note that on version LTS 1.509.3 Jenkins works stable even on high environment without any memory leak... (See attach files: "Good_GC_A1.509.3" and "Good_GC_B1.509.3") but unfortunately there is a BIG unsolved problem/bug in this version, I can't rename jobs (Deadlock! which solved on the next version LTS 1.509.4/1.532.1 that I can't use because of the memory leak).</p>

<p>TEST #4 with LTS 1.532.1:<br/>
Same issue! Jenkins stuck with 100% memory usage after only 12 hours!</p>

<p>Thank You,<br/>
Ronen.</p>

**Attachments:**

- [Good_GC_A1.509.3.JPG](https://issues.jenkins.io/secure/attachment/24688/Good_GC_A1.509.3.JPG)
- [Good_GC_B1.509.3.jpg](https://issues.jenkins.io/secure/attachment/24766/Good_GC_B1.509.3.jpg)
- [Monitor_After_12_Hours.jpg](https://issues.jenkins.io/secure/attachment/24682/Monitor_After_12_Hours.jpg)
- [Monitor_After_1_Hour.jpg](https://issues.jenkins.io/secure/attachment/24681/Monitor_After_1_Hour.jpg)
- [Monitor_After_20_Hours.jpg](https://issues.jenkins.io/secure/attachment/24683/Monitor_After_20_Hours.jpg)
- [Monitor_After_23_Hours.JPG](https://issues.jenkins.io/secure/attachment/24684/Monitor_After_23_Hours.JPG)
- [Monitor_Memory_Over_Time_Manual_GC.jpg](https://issues.jenkins.io/secure/attachment/24685/Monitor_Memory_Over_Time_Manual_GC.jpg)
- [gc.log](https://issues.jenkins.io/secure/attachment/24686/gc.log)

---
<details><summary><i>Originally reported by <a title="Ronen Peleg" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ronenpg">ronenpg</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-20620" target="_blank">Memory Leak on Jenkins LTS 1.509.4/1.532.1</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Oleg Nenashev" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>
<li><b>reported by</b>: ronenpg
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>resolution</b>: Incomplete
<li><b>resolved</b>: 2014-06-09T21:30:10+00:00
<li><b>votes</b>: 21
<li><b>watchers</b>: 15
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=elyf">elyf</a>:</i>
<p>This same problem happens with 1.540 and 1.541 on a red hat Linux platform </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ronenpg">ronenpg</a>:</i>
<p>The same problem happens with 1.532.1 LTS !</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tomerpg">tomerpg</a>:</i>
<p>The same problem with memory leak on our system with 1.509.4 LTS, <br/>
After 8 hours the Jenkins become very slowly and then stuck, I must do restart to Jenkins every day!</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vjuranek">vjuranek</a>:</i>
<p>Hi,<br/>
any change to see some heap dump (preferably some small one), or, as a first hint (as I won't be probably able to parse 40GB heap dump on my machine anyway), see output of</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-none">jmap -histo:live $JENKINS_PID |head -n 30
</pre>
</div></div>
<p>?<br/>
Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ronenpg">ronenpg</a>:</i>
<p>Sorry "vjuranek" already returns to the old and stable LTS version: 1.509.3 - The version LTS 1.509.3 come with a lots known bugs but with free of memory leaks. </p>

<p>Any Jenkins system with more than 50 jobs can see very fast the memory leaks on versions LTS 1.509.4 or 1.532.1, I don't understand why Jenkins owners ignore this very blocked issue... maybe Jenkins support team need a Load Software to find their own bugs and issues.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vjuranek">vjuranek</a>:</i>
<p>It seems that JoJ <span class="error">&#91;1&#93;</span> runs without any problems, so this memory leak maybe appears only on some specific configurations (or somehow fixed in 1.539?). Did you only upgrade Jenkins core or did you also upgrade any plugins (so we can exclude memory leak in the plugins)? By system with 50 jobs you mean 50 jobs present or 50 jobs run in parallel most of the time?<br/>
Thanks</p>

<p>Btw: there's no Jenkins support team, it's community project based on volunteers, if you want a support team, you should check Cloudbees or some other provider which offers support for Jenkins. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p><span class="error">&#91;1&#93;</span> <a href="https://ci.jenkins-ci.org/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins-ci.org/</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ronenpg">ronenpg</a>:</i>
<p>Thank You for your help <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>First with all the respect to JoJ they doesn't have the system load that we are facing here and has you can see it's snapshots version. Anyway, I didn't try the release version 1.539... BTW: In order to solve memory leaks you must use load software such as Load Runner.</p>

<p>I tried the new LTS versions after update plugins and without update plugins, still the same issue same bug and as I wrote this bug doesn't happens on previous version at all.</p>

<p>Of course it community, open source and all of that... but don't forget there is also Enterprise edition that including "Jenkins support team" and I just wonder why they not solving this super critical issue but I guess this all another story...</p>
